pulp_rpm/app/serializers.py

```diff
@@ -250,7 +250,8 @@ def deferred_validate(self, data):
         except OSError:
             raise NotAcceptable(detail='RPM file cannot be parsed for metadata.')
 
-        package = Package.objects.filter(**new_pkg)
+        attrs = {key: new_pkg[key] for key in Package.repo_key_fields}
+        package = Package.objects.filter(**attrs)
```

Comment:
Why do you filter by repo_key_fields? Such filter can return multiple packages.

Reply:
with S3, the field `time_file` changes, so filtering by all fields `package.exists()` would return `False`. I used `repo_key_fields` because I thought it assures not having duplicated packages

Reply:
For uniqueness, check unique_together on the models.
For packages it will be https://github.com/pulp/pulp_rpm/blob/master/pulp_rpm/app/models/package.py#L214